src/main/java/wolox/training/models/User.java

```diff
@@ -48,6 +49,14 @@ public void addBook(Book book) throws BookAlreadyOwnedException {
     }
   }
 
+  public void removeBook(Book book) {
+    if(!this.books.contains(book)) {
+      throw new BookNotFoundException();
+    } else {
+      this.books.remove(book);
+    }
+  }
```

Comment:
`contains` method internally use `equals` method to evaluate objects, so you need to override this. Another more elegant way could be:

`this.books.removeIf(b->b.getId()==book.getId());`